odoo/orm/models.py

```diff
@@ -5574,7 +5574,11 @@ def browse(self, ids: int | typing.Iterable[IdType] = ()) -> Self:
     @property
     def ids(self) -> list[int]:
         """ Return the list of actual record ids corresponding to ``self``. """
-        return list(origin_ids(self._ids))
+        ids = self._ids
+        if not ids or ids[0]:
```

Comment:
You normally shouldn't mix real and new records, but that it not guaranteed.  Therefore the correct way is
```suggestion
        if all(ids):
```